src/realm/inst_layout.inl

```diff
@@ -242,13 +251,72 @@ namespace Realm {
           it2 != field_offsets.end(); ++it2) {
         // should not have seen this field before
         assert(layout->fields.count(it2->first) == 0);
+
+        layout->idindexed_fields &=
+            (static_cast<size_t>(it2->first * field_stride) == reuse_offset);
+
         InstanceLayoutGeneric::FieldLayout &fl = layout->fields[it2->first];
         fl.list_idx = li;
         fl.rel_offset = /*group_offset +*/ it2->second + reuse_offset;
         fl.size_in_bytes = field_sizes[it2->first];
       }
     }
 
+    // Compute preferred dimension ordering for idindexed_fields
+    if(layout->idindexed_fields) {
+      layout->preferred_dim_order.clear();
```

Comment:
What does preferred_dim_order used for? What is the relationship between the dim_order of the function and preferred_dim_order?